Gem/Code/Source/Components/Multiplayer/EnergyCannonComponent.cpp

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project. For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/Component/ComponentApplicationBus.h>
+#include <AzCore/Component/TransformBus.h>
+#include <Source/Components/Multiplayer/EnergyBallComponent.h>
+#include <Source/Components/Multiplayer/EnergyCannonComponent.h>
+
+namespace MultiplayerSample
+{
+    EnergyCannonComponentController::EnergyCannonComponentController(EnergyCannonComponent& parent)
+        : EnergyCannonComponentControllerBase(parent)
+    {
+    }
+
+    void EnergyCannonComponentController::OnActivate([[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
+    {
+        m_firingEvent.Enqueue(GetFiringPeriod(), true);
+    }
+
+    void EnergyCannonComponentController::OnDeactivate([[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
+    {
+        m_firingEvent.RemoveFromQueue();
+    }
+
+    void EnergyCannonComponentController::OnFireEnergyBall()
+    {
+        // Re-using the same ball entity.
+        AZ::Entity* ball = nullptr;
+        AZ::ComponentApplicationBus::BroadcastResult(ball,
```

Comment:
any worry with ball reuse if the ball entity is moved to another server? this assumes whatever machine is controlling the cannon also has ownship on the ball right? (ie: single server)

Reply:
The ball is a child entity of the cannon. We can tie them together with hierarchy components if needed.

Reply:
But yeah, it gets pretty tricky. One has to assign the entity in the prefab but on spawn grab its NetEntityId and keep that. Then look it up based on its NetEntityId instead.